### ISSUE TITLE
allow rpaligned_realloc to accept oldsize == 0

### ISSUE THREAD

**Issue by jritts** [rampantpixels/rpmalloc#53]
_memory_reallocate allows zero to be passed for the parameter 'oldsize', however this convenience is essentially hidden by rpaligned_realloc() when alignment > 16.

I am tentatively using this change locally in order to avoid a hash lookup on the original allocation size.  If it looks correct, I'm happy to create a pull request, or feel free to just make the change if that's easier.

~~~~
		if (!(flags & RPMALLOC_NO_PRESERVE)) {
+			if (oldsize == 0)
+				oldsize = _memory_usable_size(ptr);
			memcpy(block, ptr, oldsize < size ? oldsize : size);
		}
~~~~

**Comment by MJansson** [comment]
Fixed in cce1b90

**Comment by jritts** [comment]
Awesome, thank you
